cpp/src/io/orc/reader_impl.cu

```diff
@@ -725,12 +725,16 @@ struct list_buffer_data {
 void generate_offsets_for_list(device_span<list_buffer_data const> buff_data,
                                rmm::cuda_stream_view stream)
 {
-  auto const transformer = [] __device__(list_buffer_data const list_data) {
+  std::vector<list_buffer_data> buff_data_vec(buff_data.size());
+  CUDF_CUDA_TRY(cudaMemcpyAsync(buff_data_vec.data(),
+                                buff_data.data(),
+                                sizeof(list_buffer_data) * buff_data.size(),
+                                cudaMemcpyDeviceToHost,
```

Comment:
```suggestion
                                cudaMemcpyDefault,
```

Reply:
I tried that initially and got some strange results. Perhaps there was something wrong with the intermediate case when I tested it though, let me see.

Reply:
I guess you don't need it if you avoid the H2D2H round-trip :)

Reply:
Yeah haha this code is gone altogether now.